### PR TITLE
Add quality score binner

### LINKED ISSUE #1462: Add quality score binner
- [ ] Should support lumping quality scores into a set number of bins.
- [ ] We should characterize the impact of quality score binning on disk space.

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #1462. WIP. Still need to:

- [x] Add docs
- [x] Characterize
- [x] Add to fragment path

### CODE REVIEW

**Comment by fnothaft** [comment]
Jenkins, retest this please.

File: adam-cli/src/main/scala/org/bdgenomics/adam/cli/Transform.scala [hunk 1]
```diff
@@ -115,6 +115,8 @@ class TransformArgs extends Args4jBase with ADAMSaveAnyArgs with ParquetArgs {
   var mdTagsFragmentSize: Long = 1000000L
   @Args4jOption(required = false, name = "-md_tag_overwrite", usage = "When adding MD tags to reads, overwrite existing incorrect tags.")
   var mdTagsOverwrite: Boolean = false
+  @Args4jOption(required = false, name = "-bin_quality_scores", usage = "Rewrites quality scores of reads into bins. User must provide bin description string.")
```
Comment (heuermh) [inline adam-cli/src/main/scala/org/bdgenomics/adam/cli/Transform.scala]: How about
Rewrites quality scores of reads into bins from a string of bin descriptions, e.g. 0,20,10;20,40,30.

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDD.scala [hunk 2]
```diff
@@ -974,4 +974,36 @@ case class AlignmentRecordRDD(
...
+   * @param bins The bins to use.
+   * @return Reads whose quality scores are binned.
+   */
+  def binQualityScores(bins: Seq[QualityScoreBin]): AlignmentRecordRDD = {
+    require(bins.nonEmpty, "Bins cannot be empty.")
+
+    // if we have multiple bins, validate them
+    // - check that we don't have gaps between bins
+    // - check that we don't have overlapping bins
+    if (bins.size > 1) {
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDD.scala]: would this check be better in `QualityScoreBin.apply`?

**Comment by heuermh** [comment]
I'd be ok merging this with the doc string fix suggested above

**Comment by fnothaft** [comment]
@heuermh addressed the doc issue. This is good to go from my side.

**Comment by heuermh** [comment]
Thank you, @fnothaft

**Pull request merged**